music_this_week_app/views.py

```diff
@@ -79,23 +80,35 @@ def callback(request):
 
 def setup(request):
     """Displays search args and a search button, which links to /search"""
-
+    #get user ip address to find user location 
+    x_forwarded_for = request.META.get('HTTP_X_FORWARDED_FOR')
+    if x_forwarded_for:
+        ip = x_forwarded_for.split(',')[0]
+    else:
+        ip = request.META.get('REMOTE_ADDR')
+    #handle invalid IP or ip == 127.0.0.1 because running locally 
```

Comment:
you mean ```#get user ip address to find user location``` ? If so the next 5 lines

```#handle invalid IP or ip == 127.0.0.1 because running locally ```
applies to the try except below